### PR TITLE
menu: make sure set_item_on is always highlighted

### LINKED ISSUE #1795: Setup-menu items aren't highlighted properly in a specific case
_Tested in the [latest master artifact (Win-64)](https://github.com/fabiangreffrath/woof/actions/runs/9939286691)._

Steps to reproduce:
- Enter a setup menu, e.g. _General_
- Optionally, move to some item with the keyboard, e.g. _Fullscreen_ (we'll call it "keyboard item" for convenience)
- Move the mouse; the keyboard item should lose the highlight
- Press a random key, e.g. `C`

You should see the keyboard item becoming "half-selected": the blinking arrows will appear besides it, but it won't get the bright highlight. This seems to be just a cosmetic error, since the item can still be interacted with normally, i.e. pressing `Enter` will work as usual.

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #1795

### CODE REVIEW

File: src/mn_setup.c [hunk 1]
```diff
@@ -3564,6 +3564,11 @@ boolean MN_SetupResponder(menu_action_t action, int ch)
         }
     }
 
+    if (highlight_item != set_item_on)
+    {
+        current_menu[set_item_on].m_flags |= S_HILITE;
+    }
```
Comment (rfomin) [inline src/mn_setup.c:3570]: But in the same function we have this code:
https://github.com/fabiangreffrath/woof/blob/d8eac1c38ba1b53bf66a7838f14bbbcf4e1416e0/src/mn_setup.c#L3446-L3449

To be honest, I don't remember the highlight_item logic. I will look into it later.

Comment (fabiangreffrath) [reply src/mn_setup.c:3570]: Yes, that part removes the highlighting from the `highlight_item` item, whereas my added code sets it for the `highlight_item` item.

Comment (rfomin) [reply src/mn_setup.c:3570]: I thought that maybe this should be fixed in the `MN_ResetMouseCursor` function. I'll check it out.

Comment (fabiangreffrath) [reply src/mn_setup.c:3570]: > I'll check it out.

Okay!

**Comment by rfomin** [comment]
Fixed in https://github.com/fabiangreffrath/woof/commit/1a3addd0d97eb7bdee9e3a93632c17c1f01d4210

**Pull request closed without merging**